### PR TITLE
fix send payments to destination address without backup

### LINKED ISSUE #3878: Bug: Send to other copay wallet that backup is not completed
Hi,
Reproducing:
1- You have a copay wallet (wallet 1) and then create another copay wallet (wallet 2).  
2- You select wallet 1 and go to send, then click on wallet icon. 
3- Now on my wallets tab, show the option to send bitcoins to wallet 2, even if wallet 2 has no backup  completed.

### PR DESCRIPTION
**Pull request by JDonadio**
Fixes #3878

### CODE REVIEW

**Comment by gabrielbazan7** [comment]
ACK!

File: src/js/controllers/index.js [hunk 1]
```diff
@@ -140,16 +140,19 @@ angular.module('copayApp.controllers').controller('indexController', function($r
...
+          });
+        } else {
+          storageService.getBackupFlag(self.walletId, function(err, val) {
             self.needsBackup = self.network == 'testnet' ? false : !val;
-          self.openWallet();
-        });
+          });
+        }
       }
+      self.openWallet();
```
Comment (cmgustavo) [inline src/js/controllers/index.js]: `openWallet()` will be opened before the other async functions. I 'm not sure it 's a good practice. `openWallet()` should be inside these async functions

**Comment by JDonadio** [comment]
Updated.

File: src/js/controllers/index.js [hunk 2]
```diff
@@ -138,21 +138,28 @@ angular.module('copayApp.controllers').controller('indexController', function($r
...
-          self.openWallet();
-        });
-      }
+      });
     });
   };
 
+  self.isBackupNeeded = function(fc, cb) {
+    if (fc.isPrivKeyExternal()) return cb(false);
+
```
Comment (matiu) [inline src/js/controllers/index.js]: from here it should be something like:

if (!fc.cretentials.mnemonic) return cb(false);
if (... testnet) return cb(false)

... getBackupFlag

File: src/js/controllers/index.js [hunk 3]
```diff
@@ -138,21 +138,28 @@ angular.module('copayApp.controllers').controller('indexController', function($r
...
-          else
-            self.needsBackup = self.network == 'testnet' ? false : !val;
-          self.openWallet();
-        });
-      }
+      });
     });
   };
 
+  self.isBackupNeeded = function(fc, cb) {
```
Comment (matiu) [inline src/js/controllers/index.js]: should be in profileService.

File: src/js/controllers/walletHome.js [hunk 4]
```diff
@@ -243,7 +244,11 @@ angular.module('copayApp.controllers').controller('walletHomeController', functi
             return;
           }
 
-          $modalInstance.close(addr);
+          storageService.getBackupFlag(walletId, function(err, val) {
```
Comment (matiu) [inline src/js/controllers/walletHome.js]: should call isBackupNeeded

**Comment by JDonadio** [comment]
Updated

**Comment by JDonadio** [comment]
Updated.

**Comment by JDonadio** [comment]
Rebased

**Comment by cmgustavo** [comment]
LGTM

**Pull request merged**